src/lib.rs

```diff
@@ -553,6 +606,133 @@ impl CompileOptions {
         }
     }
 
+    /// Sets the callback for handling the `#include` directive.
+    ///
+    /// The arguments to the callback are the name of the source being requested,
+    /// the type of include directive (`Relative` for `#include "foo"`, `Standard` for `#include <foo>`),
+    /// the name of the source containing the directive and the current include depth from the original
+    /// source.
+    ///
+    /// The return value of the callback should be `Ok` if the source was successfully found,
+    /// and an `Err` containing some suitable error message to display otherwise.
+    ///
+    /// Behaviour note: If `Err` is returned for a `Relative` include request, the callback will be
+    /// tried again with `Standard`, which is similar to include directive behaviour in C.
+    pub fn set_include_callback<F>(&mut self, f: F)
+        where F : Fn(&str, IncludeType, &str, usize) -> result::Result<ResolvedInclude, String> + 'a
+    {
+        use std::ffi::{CString, CStr};
```

Comment:
Already `use`d at the beginning of the file?

Reply:
Yep, good spot.